lib/util/http/index.ts

```diff
@@ -28,6 +32,14 @@ export interface HttpResponse<T = string> {
   headers: any;
 }
 
+function cloneResponse(response: any): any {
+  // clone body and headers so that the cached result doesn't get accidentally mutated
+  return {
+    body: clone(response.body),
+    headers: clone(response.headers),
+  };
+}
+
```

Comment:
```suggestion
function cloneResponse<T>(response: GotPromise<T>): HttpResponse<T> {
  // clone body and headers so that the cached result doesn't get accidentally mutated
  return {
    body: clone(response.body),
    headers: clone(response.headers),
  };
}

```

Reply:
Isn't it taking a resolved GotPromise and not a raw GotPromise?